src/extra/AutoUI/Filters/FocusSearch.tsx

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import { Txt } from "../../../components/Txt";
+import { Checkbox } from "../../../components/Checkbox";
+import reject from "lodash/reject";
+import groupBy from "lodash/groupBy";
+import styled from "styled-components";
+import { Box } from "../../../components/Box";
+import { Flex } from "../../../components/Flex";
+import { useHistory } from "../../../hooks/useHistory";
+import { AutoUIContext } from "../schemaOps";
+import { stopEvent } from "../../../utils";
+import { AutoUIModel } from "..";
+
+const Focus = styled(Box)`
+	flex-basis: 500px;
+	background-color: white;
+	border: solid 1px ${(props) => props.theme.colors.quartenary.dark};
+	max-height: 200px;
+	position: absolute;
+	width: 100%;
+	z-index: 1;
+	border-radius: 0 0 ${(props) => props.theme.global.drop.border.radius}
+		${(props) => props.theme.global.drop.border.radius};
+	overflow: hidden;
+`;
+
+const FocusContent = styled(Box)`
+	max-height: 180px;
+	overflow-y: auto;
+	overflow-x: auto;
+`;
+
+const FocusItem = styled(Box)<{ hasGetBaseUrl: boolean }>`
+	cursor: ${(props) => (props.hasGetBaseUrl ? "pointer" : "default")};
+	&:hover {
+		background: #dde1f0; // This is the background color Select uses for entities on hover. We do not have it in our theme
+	}
+`;
+
+interface FocusSearchProps<T extends { id: number; [key: string]: any }> {
```

Comment:
I think that this can also be done like:

```suggestion
interface FocusSearchProps<T extends { id: number; searchTerms: string[] }> {
```

Reply:
I don't think so, I don't recall exactly but judging by what happens when I accept your suggestion, I think `[key: string]` is meant to be any key in the data object and `any` is to allow the value of the key to be any type (string, boolean, array)

Reply:
🤔 ok let's leave it like this and I'll fix the typing as soon as I can